packages/zoltan2/src/algorithms/color/Zoltan2_AlgHybridGMB.hpp

```diff
@@ -0,0 +1,458 @@
+      mapOwned = rcp(new map_t(dummy, vtxIDs, 0, comm));
+
+      dummy = Teuchos::OrdinalTraits <Tpetra::global_size_t>::invalid();
+      mapWithCopies = rcp(new map_t(dummy, 
+                                Teuchos::arrayViewFromVector(ownedPlusGhosts),
+                                0, comm)); 
+                                      
+      //create the FEMultiVector for the distributed communication.
+      //We also use the views from this datastructure as arguments to
+      //KokkosKernels coloring functions.
+      typedef Tpetra::Import<lno_t, gno_t> import_t;
+      Teuchos::RCP<import_t> importer = rcp(new import_t(mapOwned, 
+                                                            mapWithCopies));
+      Teuchos::RCP<femv_t> femv = rcp(new femv_t(mapOwned, 
+                                                    importer, 1, true));
+      timeReorder->stop();
+      //Get color array to fill
+      ArrayRCP<int> colors = solution->getColorsRCP();
+      for(size_t i=0; i<nVtx; i++){
+        colors[i] = 0;
+      } 
+      
+      //Create random numbers seeded on global IDs so that we don't
+      //need to communicate for consistency. These numbers determine
+      //which vertex gets recolored in the event of a conflict.
+      //taken directly from the Zoltan coloring implementation 
+      std::vector<int> rand(finalGIDs.size());
+      for(size_t i = 0; i < finalGIDs.size(); i++){
+        Zoltan_Srand((unsigned int) finalGIDs[i], NULL);
+        rand[i] = (int) (((double) Zoltan_Rand(NULL)/(double) ZOLTAN_RAND_MAX)*100000000);
+      }
+
+      
+      ArrayView<const offset_t> finalOffsets = Teuchos::arrayViewFromVector(finalOffset_vec);
+      ArrayView<const lno_t> finalAdjs = Teuchos::arrayViewFromVector(finalAdjs_vec);
+
+      // call coloring function
+      hybridGMB(nVtx, nInterior, finalAdjs, finalOffsets,colors,femv,finalGIDs,rand,timeInterior,timeBoundary);
+      
+      //copy colors to the output array.
+      for(int i = 0; i < colors.size(); i++){
+        colors[reorderToLocal[i]] = femv->getData(0)[i];
+      }
+     
+      //RESULT REPORTING INSTRUMENTATION
+      //Teuchos::TimeMonitor::summarize();
+      Teuchos::TimeMonitor::zeroOutTimers();
+      comm->barrier();
+    }
+     
+    void hybridGMB(const size_t nVtx,lno_t nInterior, Teuchos::ArrayView<const lno_t> adjs, 
+                   Teuchos::ArrayView<const offset_t> offsets, 
+                   Teuchos::ArrayRCP<int> colors, Teuchos::RCP<femv_t> femv,
+                   std::vector<gno_t> reorderGIDs,
+                   std::vector<int> rand,
+                   Teuchos::RCP<Teuchos::Time> timeInterior,
+                   Teuchos::RCP<Teuchos::Time> timeBoundary){
+     Teuchos::RCP<Teuchos::Time> 
+             timeBoundaryComm(Teuchos::TimeMonitor::getNewTimer("03 BOUNDARY-COMM")),
+             timeBoundaryComp(Teuchos::TimeMonitor::getNewTimer("04 BOUNDARY-COMP"));
+      
+      timeInterior->start();
```

Comment:
Use `Teuchos::TimeMonitor` instead of calling start/stop explicitly.

Reply:
I am not certain how to achieve the same effect using Teuchos::TimeMonitor without negatively impacting readability, so I just removed the timers.